packages/flutter/lib/src/widgets/two_pane.dart

```diff
@@ -0,0 +1,314 @@
+  ///
+  /// For example, when TwoPane is the body of a Scaffold, the appbar adds extra
+  /// spacing between TwoPane and the edge of the screen. If TwoPane is used
+  /// with a [Axis.vertical] [direction], the separation between the two panes
+  /// will not align with [DisplayFeature]s unless the [padding] also includes
+  /// the height of the appbar.
+  ///
+  /// Defaults to [MediaQueryData.padding].
+  final EdgeInsets padding;
+
+  TextDirection? _getTextDirection(BuildContext context) =>
+      textDirection ?? Directionality.maybeOf(context);
+
+  @override
+  Widget build(BuildContext context) {
+    final MediaQueryData? mediaQuery = MediaQuery.maybeOf(context);
+    final DisplayFeature? displayFeature = _separatingDisplayFeature(mediaQuery);
+
+    late Axis _direction;
+    late int pane1Flex;
+    late int pane2Flex;
+    late Widget _pane1;
+    late Widget _pane2;
+    late Widget _delimiter;
+    final TextDirection? _textDirection = _getTextDirection(context);
+    const int fractionBase = 1000000000000;
+
+    if (mediaQuery == null || displayFeature == null) {
+      // The display is continuous, nothing is overridden.
+      _direction = direction;
+      pane1Flex = (fractionBase * paneProportion).toInt();
+      pane2Flex = fractionBase - pane1Flex;
+      _pane1 = pane1;
+      _pane2 = pane2;
+      _delimiter = Container();
+    } else {
+      // The display has a seam that splits it in two panels.
+      // final EdgeInsets padding = this.padding;
+      final Size size = Size(
+          mediaQuery.size.width - padding.horizontal,
+          mediaQuery.size.height - padding.vertical);
+      final Rect seam = displayFeature.bounds.translate(-padding.left, -padding.top);
+
+      if (seam.width < seam.height) {
+        // Seam is tall. Panels are left and right.
+        _direction = Axis.horizontal;
+        _delimiter = Container(width: seam.size.width);
+        assert(_textDirection != null);
+        final int leftPane = (seam.left * fractionBase).toInt();
+        final int rightPane = ((size.width - seam.right) * fractionBase).toInt();
+        if (_textDirection == TextDirection.ltr) {
+          pane1Flex = leftPane;
+          pane2Flex = rightPane;
+        } else {
+          pane1Flex = rightPane;
+          pane2Flex = leftPane;
+        }
+      } else {
+        // Seam is wide. Panels are above and below.
+        _direction = Axis.vertical;
+        _delimiter = Container(height: seam.size.height);
+        final int topPane = (seam.top * fractionBase).toInt();
+        final int bottomPane = ((size.height - seam.bottom) * fractionBase).toInt();
+        if (verticalDirection == VerticalDirection.down) {
+          pane1Flex = topPane;
+          pane2Flex = bottomPane;
+        } else {
+          pane1Flex = bottomPane;
+          pane2Flex = topPane;
+        }
+      }
+    }
+
+    if (mediaQuery==null || panePriority != TwoPanePriority.both) {
+      _pane1 = pane1;
```

Comment:
`_pane1` and `_pane2` will already be assigned when `mediaQuery == null` because of the first if (line 133)